Add Results notebook

cjcardinale opened the pull request:
A simple notebook that downloads regridded CMIP6 sea ice data from a new S3 bucket. Instead of one zarr file for each grid, each model is uploaded as a separate zarr file and opened with `xr.open_mfdataset(fileset, engine='zarr', consolidated=True)`. 

Future updates will include: 
- region and perennial sea ice masking
- MAE/SAE 
- spatial averages
- at least one figure from the paper

cjcardinale committed:
Add Results notebook

Simple notebook that downloads regridded CMIP6 sea ice data from S3 bucket

cjcardinale committed:
Plots some figures from the paper

also adds code that applies perennial mask and computes ensemble means, spatial means, internal variability, MAE, and SAE

cjcardinale commented:
The new commit includes the updates described above. I may need to change the chunk sizes for better performance and change the data types from float16 to float32 or float64 to avoid RuntimeWarnings (more comments in the notebook).

akpetty commented:
Awesome, this seems to run well. I would suggest adding some docstrings and more description to just make a bit clearer what the functions are doing?! I think AI tools are quite helpful for this now!

cjcardinale committed:
Update results notebook and add docstrings to functions

cjcardinale committed:
Move is_valid_zarr_store to functions.py

cjcardinale committed:
Add import zarr to functions.py

cjcardinale commented:
The latest commits add plotting code for most Figures (excluding many supplemental figures). I've uploaded 17 models to the cloud so far. I last ran the notebook with 3 models, which took around 7 minutes. So it should take around 1.5 hours when loading and processing all 42 models. I also added docstrings to every function.

akpetty commented:
Thanks Chris. A few things:

1. It would be great to have a quick test of running this locally as that' show I expect most potential users will engage with this. Must be somthing somewhere with the cryocloud env we can export as a requirements file? I think uv might be a good idea as that's super quick, although I am having issues resolving dependencies etc and getting it to run. To get started:
```
curl -LsSf https://astral.sh/uv/install.sh | sh
uv venv
source .venv/bin/activate
uv pip install 
```
 

2. Can you include in the readme info about environments and also some thoughts about the time it takes to run with X number of models for different cryocloud compute choices vs locally? I think that would be very nice to have as a reference. 

3. For the notebook, it's pretty trivial now to ask cursor or VS Code (and i'm guessing CHatGPT directly) to add comments to notebooks. I asked it to add comments for each cell describing in a line or two what it does and also to break it into sections with headers and description in markdown. I think best for you to do this as you know the code more than me and can verify that output! I will check through and make sure it's logical!

akpetty commented:
And FYI this did run for me on cryocloud, about 7/8 minutes total I think with 3 CPU server.

cjcardinale committed:
Update README.md

cjcardinale committed:
Update README.md

cjcardinale committed:
Update README.md

cjcardinale committed:
Update results notebooks

Adds a fast results notebook and updates the data processing and results notebook. Also adds environment.yml and requirements.txt files.

cjcardinale committed:
small readme edit

cjcardinale committed:
small readme edit

cjcardinale commented:
- Latest commits add a notebook (Results.ipynb) that loads the climatological datasets for faster plotting. The data processing notebook is now called Data-processing_results.ipynb and still plots the results. 

- Now includes requirements.txt and environment.yml files, but I still haven't tested them. 

-  Added a readme that describes the notebooks and includes a section on model changes.

cjcardinale committed:
Update README.md

cjcardinale committed:
Update README.md

cjcardinale committed:
Update requirements.txt and environment.yml

Also adds a cell at the top of each notebook that correctly changes the PROJ_LIB variable and the PyProj data directory when using conda/mamba to create an envronment

akpetty commented:
Really nice work!! A few little things:

I think you need to specify the Python version, this worked for me: uv venv --python 3.11

I would suggest calling them 'CMIP6_seaice_preprocessed_data_fast.ipynb; and 'CMIP6_seaice_inc_full_data_processing.ipynb'

I would also suggest Notebook titles more like Constraining CMIP6 sea ice simulations with ICESat-2 (preprocessed data, fast analysis) and Constraining CMIP6 sea ice simulations with ICESat-2 (inc data processing, slow)

cjcardinale committed:
Update README.md

change notebook titles and update pip setup

cjcardinale committed:
Update README.md

updated repository structure

cjcardinale committed:
Renamed notebooks for clarity

Also removed seaborn from the pip install/upgrade cell as the latest version is in cryocloud by default

cjcardinale committed:
rename notebook

removed extra .ipynb in the file name

cjcardinale committed:
markdown edit
